doc-content/jbpm-docs/src/main/asciidoc/CoreEngine/service-runtime-data-con.adoc

```diff
@@ -35,94 +35,533 @@ List<tasksummary> taskSummaries = runtimeDataService.getTasksAssignedAsPotential
+     *         the given criteria (states and initiator).
+     */
     Collection<ProcessInstanceDesc> getProcessInstances(List<Integer> states, String initiator, QueryContext queryContext);
 
+    /**
+     * Returns a list of process instance descriptions found for the given process ID and statuses and initiated by <code>initiator</code>
+     * @param states A list of possible state (int) values that the {@link ProcessInstance} can have.
+     * @param processId The ID of the {@link Process} (definition) used when starting the process instance.
+     * @param initiator The initiator of the {@link ProcessInstance}.
+     * @param queryContext control parameters for the result, such as sorting and paging
+     * @return A list of {@link ProcessInstanceDesc} instances representing the process instances that match
+     *         the given criteria (states, processId, and initiator).
+     */
     Collection<ProcessInstanceDesc> getProcessInstancesByProcessId(List<Integer> states, String processId, String initiator, QueryContext queryContext);
 
+    /**
+    * Returns a list of process instance descriptions found for the given process name and statuses and initiated by <code>initiator</code>
+     * @param states A list of possible state (int) values that the {@link ProcessInstance} can have.
+     * @param processName The name (not the ID) of the {@link Process} (definition) used when starting the process instance.
+     * @param initiator The initiator of the {@link ProcessInstance}.
+     * @param queryContext control parameters for the result, such as sorting and paging
+     * @return A list of {@link ProcessInstanceDesc} instances representing the process instances that match
+     *         the given criteria (states, processName and initiator).
+     */
     Collection<ProcessInstanceDesc> getProcessInstancesByProcessName(List<Integer> states, String processName, String initiator, QueryContext queryContext);
 
+    /**
+     * Returns a list of process instance descriptions found for the given deployment ID and statuses.
+     * @param deploymentId The deployment ID of the runtime.
+     * @param states A list of possible state (int) values that the {@link ProcessInstance} can have.
+     * @param queryContext control parameters for the result, such as sorting and paging
+     * @return A list of {@link ProcessInstanceDesc} instances representing the process instances that match
+     *         the given criteria (deploymentId and states).
+     */
     Collection<ProcessInstanceDesc> getProcessInstancesByDeploymentId(String deploymentId, List<Integer> states, QueryContext queryContext);
 
+    /**
+     * Returns process instance descriptions found for the given processInstanceId. If none are found, returns null. At the same time the method
+     * fetches all active tasks (in status: Ready, Reserved, InProgress) to provide the information about what user task is keeping each instance
```

Comment:
Try to avoid the parenthesis here. 
what user task is keeping each instance and who owns them (if were already claimed).
What is 'them'? The instances or the user tasks? You have a singular/plural issue. Each/them. Not sure the best way to fix as I don't know the meaning.